vulnfeeds/git/repository.go

```diff
@@ -8,31 +8,46 @@ import (
 	"github.com/go-git/go-git/v5/config"
 	"github.com/go-git/go-git/v5/plumbing/transport"
 	"github.com/go-git/go-git/v5/storage/memory"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/google/osv/vulnfeeds/cves"
 )
 
-// A version holds a tag and corresponding commit hash.
-type Version struct {
+// A GitTag holds a Git tag and corresponding commit hash.
+type Tag struct {
 	Tag    string // Git tag
 	Commit string // Git commit hash
 }
 
-// Versions is an ordered array of Version.
-type Versions []Version
+type Tags []Tag
 
-func (v Versions) Len() int           { return len(v) }
-func (v Versions) Less(i, j int) bool { return v[i].Tag < v[j].Tag }
-func (v Versions) Swap(i, j int)      { v[i], v[j] = v[j], v[i] }
+func (t Tags) Len() int           { return len(t) }
+func (t Tags) Less(i, j int) bool { return t[i].Tag < t[j].Tag }
+func (t Tags) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
+
+type NormalizedTag struct {
+	OriginalTag string
+	Commit      string
+}
 
 // RepoTagsMap acts as a cache for RepoTags results, keyed on the repo's URL.
-type RepoTagsMap map[string]Versions
+// repo URL -> Tag -> GitTag
+type RepoTagsMap struct {
+	Tag           map[string]Tag
```

Comment:
nit: can you add some short comments on the expected key for these? It's not obvious from the code itself.

Reply:
Improved in [95c7d05](https://github.com/google/osv.dev/pull/1057/commits/95c7d0521ed2ea24906a34903e6f2469d4fead2d)